Bad C++ codegen on distinct generics C++ types (Atomics here).

I expect this would happen as well with distinct std::vector.

Testcase:

```Nim
import std/atomics
type Futex = distinct Atomic[int32]

var x: Futex
```

Output
```
/home/beta/.cache/nim/cpp_futex_d/@mcpp_futex.nim.cpp:31:1: error: invalid use of template-name ‘std::atomic’ without an argument list
   31 | std::atomic x__bqCKmFWuGC2fnFKyiPKa3A;
      | ^~~
/home/beta/.cache/nim/cpp_futex_d/@mcpp_futex.nim.cpp:31:1: note: class template argument deduction is only available with ‘-std=c++17’ or ‘-std=gnu++17’
In file included from /usr/include/c++/9.2.0/atomic:41,
                 from /home/beta/.cache/nim/cpp_futex_d/@mcpp_futex.nim.cpp:10:
/usr/include/c++/9.2.0/bits/atomic_base.h:145:12: note: ‘template<class _Tp> struct std::atomic’ declared here
  145 |     struct atomic;
      |            ^~~~~~
```

So it seems like with distinct the generic type gets stripped out during codegen.

--- a/compiler/ccgtypes.nim
+++ b/compiler/ccgtypes.nim
@@ -826,7 +826,8 @@ proc getTypeDescAux(m: BModule, origTyp: PType, check: var IntSet; kind: TSymKin
       m.s[cfsTypes].addf("typedef $1 $2[$3];$n",
            [foo, result, rope(n)])
   of tyObject, tyTuple:
-    if isImportedCppType(t) and origTyp.kind == tyGenericInst:
+    let tt = origTyp.skipTypes({tyDistinct})
+    if isImportedCppType(t) and tt.kind == tyGenericInst:
       let cppNameAsRope = getTypeName(m, t, sig)
       let cppName = $cppNameAsRope
       var i = 0
@@ -849,7 +850,7 @@ proc getTypeDescAux(m: BModule, origTyp: PType, check: var IntSet; kind: TSymKin
             result.add cppName.substr(chunkStart, chunkEnd)
             chunkStart = i
 
-            let typeInSlot = resolveStarsInCppType(origTyp, idx + 1, stars)
+            let typeInSlot = resolveStarsInCppType(tt, idx + 1, stars)
             addResultType(typeInSlot)
         else:
           inc i
@@ -858,9 +859,9 @@ proc getTypeDescAux(m: BModule, origTyp: PType, check: var IntSet; kind: TSymKin
         result.add cppName.substr(chunkStart)
       else:
         result = cppNameAsRope & "<"
-        for i in 1..<origTyp.len-1:
+        for i in 1..<tt.len-1:
           if i > 1: result.add(" COMMA ")
-          addResultType(origTyp[i])
+          addResultType(tt[i])
         result.add("> ")
       # always call for sideeffects:
       assert t.kind != tyTuple

--- a/tests/cpp/t12946.nim
+++ b/tests/cpp/t12946.nim
@@ -1,0 +1,4 @@
+import std/atomics
+type Futex = distinct Atomic[int32]
+
+var x: Futex

